docs/scanners/sgrep.md

```diff
@@ -0,0 +1,18 @@
+# [sgrep](https://sgrep.dev)
+
+This scanner can flag anti-patterns found in a codebase or require that certain strings be present. This might be useful for preventing the use of dangerous methods like `eval()` in Ruby (which might allow for RCE) or `dangerouslySetInnerHTML` in React (which might allow for XSS). By default, all found patterns are added to the info section of the report. If a found pattern is forbidden, this scanner will fail and the `message` will be show to the developer in the report to give additional context on why this was an issue. A `required` pattern must be found in order for the scan to pass.
```

Comment:
[sgrep](https://sgrep.dev/) (syntactic grep) is an open-source tool for finding patterns in code. It lifts grep-esque patterns into AST matchers. Compared to regexes these patterns aren’t affected by whitespaces, comments, newlines, the order of keyword arguments, variable renaming, and other language nuances.

This scanner is useful for preventing the use of known anti-patterns in a codebase or enforcing the correct use of secure-by-default frameworks (i.e. `eval()` in Ruby which might allow for RCE, or `dangerouslySetInnerHTML` in React which might allow for XSS). 

If a found pattern is forbidden this scanner will fail and the `message` will be show to the developer in the report. A `required` pattern must be found in order for the scan to pass.

Reply:
@pablo Thoughts on above?